test/token.test.ts

```diff
@@ -261,30 +349,30 @@ describe('token integration', () => {
         context.tokenA.approveBase(AccountUpdateForest.fromFlatArray([
```

Comment:
I wonder if we can move the `approveBase` call to a third party contract methods themselves.

Reply:
I think we could do that if we passed in the `AccountUpdate`s from the other contract. So we would have something like

```TypeScript
withdraw(amount: UInt64): AccountUpdate
deposit(incomingAccountUpdate: AccountUpdate): AccountUpdate
```

That way, we could move the call to `approveBase` to `deposit`. But it would be less flexible. For example, you couldn't create a transaction that collects tokens from two accounts and sends them to one, or things like that.
Maybe we could mention different approaches in the tutorial?